common/errors.js

```diff
@@ -125,26 +133,32 @@
         // TODO: implement hierarchies of exceptions in ermrestJS and use that hierarchy to conditionally check for certain exceptions
         function handleException(exception) {
             $log.info(exception);
+            
             var reloadCb = function() {
                 window.location.reload();
             }; 
             if (exceptionFlag || window.location.pathname.indexOf('/search/') != -1 || window.location.pathname.indexOf('/viewer/') != -1) return;
 
             if (ERMrest && exception instanceof ERMrest.UnauthorizedError || exception.code == errorNames.unauthorized) {
                 Session.loginInAPopUp(reloadCb);
-            } else {
+            }
+            else if (exception.code && exception.code.indexOf(messageMap.multipleDataErrorCode) !== -1){
```

Comment:
This `exception.code` is usually a number (409, 401, etc.). Since these `multipleRecordError` and `noRecordError` are custom errors that we made, we either have to change the code to something meaningful (an actual error code number) and then add `sub-code`, or we can accept the code to be string in some cases.

In either cases, this check `exception.code.indexOf(messageMap.multipleDataErrorCode) !== -1` can cause problem, change it to  `exception.code == errorNames.multipleRecords`.